finetune/util/huggingface_interface.py

```diff
@@ -0,0 +1,200 @@
+import h5py
+import numpy as np
+import joblib as jl
+import os
+import tqdl
+import tensorflow as tf
+from tensorflow.python.keras.saving.hdf5_format import load_attributes_from_hdf5_group
+from finetune.util.download import FINETUNE_BASE_FOLDER
+from finetune.util.shapes import lengths_from_eos_idx
+from finetune.encoding.input_encoder import BaseEncoder
+from finetune.encoding.input_encoder import EncodedOutput
+from finetune.base_models import SourceModel
+from finetune.optimizers.recompute_grads import recompute_grad
+
+from tensorflow.python.util import tf_inspect
+
+
+def load_weights_from_hdf5_group_by_name(filepath, weights_replacement):
+    with h5py.File(filepath, "r") as f:
+        if "layer_names" not in f.attrs and "model_weights" in f:
```

Comment:
I'm guessing this is where we'll run into the most trouble w/ model support?  Models that do something outside the ordinary in serialization?